swagger-templates/src/main/java/com/okta/swagger/codegen/AbstractOktaJavaClientCodegen.java

```diff
@@ -688,16 +687,17 @@ public CodegenModel fromModel(String name, Schema model, Map<String, Schema> all
                 if (discriminatorRoot != null) {
                     model.getExtensions().put("discriminatorRoot", discriminatorRoot);
                 }
-
-            } else if(codegenModel.parent != null) {
-                // handle correctly model inheritance described as
-                // allOf:
-                // - $ref:
-                codegenModel.parent = toApiName(codegenModel.parent);
-                codegenModel.getVendorExtensions().remove("top-level");
             }
         }
 
+        if(codegenModel.parent != null) {
+            // handle correctly model inheritance described as
+            // allOf:
+            // - $ref:
+            codegenModel.parent = toApiName(codegenModel.parent);
+            codegenModel.getVendorExtensions().remove("top-level");
```

Comment:
is there really zero APIs that are top-level somewhere and a superclass elsewhere?

Reply:
I don't believe that is true.